lib/winston/transports/file.js

```diff
@@ -42,7 +42,7 @@ var File = exports.File = function (options) {
 
   if (options.filename || options.dirname) {
     throwIf('filename or dirname', 'stream');
-    this._basename = this.filename = options.filename
```

Comment:
This change is not strictly additive. You are removing `this.filename` here, which is breaking.